core/tests/puppeteer-acceptance-tests/spec/helpers/reporter.ts

```diff
@@ -33,30 +40,37 @@ let suiteCount,
     none: '\x1B[0m'
   };
 
-let print = function() {
-  process.stdout.write(util.format.apply(this, arguments));
+let print = function(message: string): void {
+  process.stdout.write(util.format(message));
 };
 
-let printNewline = function() {
+let printNewline = function(): void {
   print('\n');
 };
 
-let colored = function(color, str) {
+let colored = function(color: keyof typeof ansi, str: string): string {
   return (ansi[color] + str + ansi.none);
 };
 
-let printLog = function(result) {
+let printFailures = function(
+    result: jasmine.JasmineDoneInfo | jasmine.SpecResult | jasmine.SuiteResult
+): void {
   for (let i = 0; i < result.failedExpectations.length; i++) {
-    const failedExpectation = result.failedExpectations[i];
     printNewline();
     print('Message:');
     printNewline();
-    print(colored('red', failedExpectation.message));
+    print(colored('red', result.failedExpectations[i].message));
     printNewline();
     print('Stack:');
     printNewline();
-    print(!failedExpectation.stack ? '' : failedExpectation.stack);
+    print(
+      !result.failedExpectations[i].stack ? '' :
+      result.failedExpectations[i].stack);
```

Comment:
Just wondering if this can this be simplified ... e.g. using nullish coalescing operator `??`.

Reply:
Done